app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -44,43 +46,52 @@ class CacheConnector @Inject()(cacheRepository: TaiCacheRepository, mongoConfig:
     Play.current.configuration.underlying).JsonCrypto
   private val defaultKey = "TAI-DATA"
 
-  def createOrUpdate[T](id: String, data: T, key: String = defaultKey)(implicit writes: Writes[T]): Future[T] = {
+  private def cacheId(nino: Nino)(implicit hc: HeaderCarrier): String = {
+    val sessionId = hc.sessionId.map(_.value).getOrElse(throw new RuntimeException("Error while fetching session id"))
```

Comment:
Do we want to throw an exception here or would it be better to just cache using the nino by itself. What scenarios will the session id not be available?